drivers/sensor/mcux_tempmon/mcux_tempmon.c

```diff
@@ -0,0 +1,118 @@
+#define OCOTP_ANA1_ROOM_COUNT_SHIFT 20
+#define OCOTP_ANA1_ROOM_COUNT_MASK  (BIT_MASK(12) << OCOTP_ANA1_ROOM_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_COUNT_SHIFT  8
+#define OCOTP_ANA1_HOT_COUNT_MASK   (BIT_MASK(12) << OCOTP_ANA1_HOT_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_TEMP_SHIFT   0
+#define OCOTP_ANA1_HOT_TEMP_MASK    (BIT_MASK(8) << OCOTP_ANA1_HOT_TEMP_SHIFT)
+
+#define TEMPMON_ROOM_TEMP 25.0f
+
+struct mcux_tempmon_data {
+	uint8_t hot_temp;
+	uint16_t hot_cnt;
+	uint16_t room_cnt;
+};
+
+struct mcux_tempmon_config {
+	TEMPMON_Type *base;
+};
+
+static int mcux_tempmon_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct mcux_tempmon_config *cfg = dev->config;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	/* Start the measure */
+	cfg->base->TEMPSENSE0 |= TEMPMON_TEMPSENSE0_MEASURE_TEMP_MASK;
+
+	return 0;
+}
+
+static int mcux_tempmon_channel_get(const struct device *dev, enum sensor_channel chan,
+				    struct sensor_value *val)
+{
+	const struct mcux_tempmon_data *data = dev->data;
+	const struct mcux_tempmon_config *cfg = dev->config;
+	uint16_t temp_cnt;
+	float temp;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	if (!(cfg->base->TEMPSENSE0 & TEMPMON_TEMPSENSE0_FINISHED_MASK)) {
+		/* Measure is not ready */
+		return -EINVAL;
+	}
+
+	temp_cnt = (cfg->base->TEMPSENSE0 & TEMPMON_TEMPSENSE0_TEMP_CNT_MASK) >>
+		   TEMPMON_TEMPSENSE0_TEMP_CNT_SHIFT;
+	temp = (float)data->hot_temp - ((float)temp_cnt - (float)data->hot_cnt) *
+					       (((float)data->hot_temp - TEMPMON_ROOM_TEMP) /
+						((float)data->room_cnt - (float)data->hot_cnt));
+
+	/* Stop the measure */
+	cfg->base->TEMPSENSE0 &= ~TEMPMON_TEMPSENSE0_MEASURE_TEMP_MASK;
+
+	return sensor_value_from_double(val, temp);
+}
+
+static const struct sensor_driver_api mcux_tempmon_driver_api = {
+	.sample_fetch = mcux_tempmon_sample_fetch,
+	.channel_get = mcux_tempmon_channel_get,
+};
+
+static int mcux_tempmon_init(const struct device *dev)
+{
+	struct mcux_tempmon_data *data = dev->data;
+	const struct mcux_tempmon_config *cfg = dev->config;
+	uint32_t ocotp_ana1;
+
+	/* Enable the temperature sensor */
+	cfg->base->TEMPSENSE0 &= ~TEMPMON_TEMPSENSE0_POWER_DOWN_MASK;
+
+	/* Single measure with no repeat mode */
+	cfg->base->TEMPSENSE1 = TEMPMON_TEMPSENSE1_MEASURE_FREQ(0);
+
+	/* Read calibration data */
+	ocotp_ana1 = sys_read32(OCOTP_ANA1_ADDR);
+	data->hot_temp = (ocotp_ana1 & OCOTP_ANA1_HOT_TEMP_MASK) >> OCOTP_ANA1_HOT_TEMP_SHIFT;
+	data->hot_cnt = (ocotp_ana1 & OCOTP_ANA1_HOT_COUNT_MASK) >> OCOTP_ANA1_HOT_COUNT_SHIFT;
+	data->room_cnt = (ocotp_ana1 & OCOTP_ANA1_ROOM_COUNT_MASK) >> OCOTP_ANA1_ROOM_COUNT_SHIFT;
+
+	return 0;
+}
+
+static struct mcux_tempmon_data mcux_tempmon_dev_data = {
+	.hot_temp = 0,
+	.hot_cnt = 0,
+	.room_cnt = 0,
+};
+
+static const struct mcux_tempmon_config mcux_tempmon_dev_config = {
+	.base = (TEMPMON_Type *)DT_REG_ADDR(DT_NODELABEL(tempmon)),
```

Comment:
I'd suggest to use `DT_INST_REG_ADDR(0)` to not depend on the node label. You already set correctly the `DT_DRV_COMPAT` at the top of this file.